core/planner/planner_test.go

```diff
@@ -173,43 +157,46 @@ func TestFlatTariffTargetInThePast(t *testing.T) {
 
 	plan, err := p.Plan(time.Hour, clock.Now().Add(30*time.Minute))
 	assert.NoError(t, err)
-	assert.True(t, !ActiveSlot(clock, plan).End.IsZero(), "should start past start time")
+	assert.True(t, !SlotAt(clock.Now(), plan).IsEmpty(), "should start past start time")
 
 	plan, err = p.Plan(time.Hour, clock.Now().Add(-30*time.Minute))
 	assert.NoError(t, err)
-	assert.False(t, !ActiveSlot(clock, plan).End.IsZero(), "should not start past target time")
+	assert.False(t, !SlotAt(clock.Now(), plan).IsEmpty(), "should not start past target time")
 }
 
 func TestFlatTariffLongSlots(t *testing.T) {
 	clock := clock.NewMock()
 	ctrl := gomock.NewController(t)
 
 	trf := mock.NewMockTariff(ctrl)
-	trf.EXPECT().Rates().AnyTimes().Return(rates([]float64{0}, clock.Now(), func(start time.Time) time.Time {
-		return start.Add(24 * time.Hour)
-	}), nil)
+	trf.EXPECT().Rates().AnyTimes().Return(rates([]float64{0}, clock.Now(), 24*time.Hour), nil)
 
 	p := &Planner{
 		log:    util.NewLogger("foo"),
 		clock:  clock,
 		tariff: trf,
 	}
 
+	// for a single slot, we always expect charging to start early
```

Comment:
Because the complete planner works in 1 hours slots now and with that context this is not a problem